src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -213,7 +216,7 @@ public void select(final int delay) {
       } catch (NullPointerException e) {
         //There is a bug in some JVMs around this where the select() can throw an NPE from native code.
       }
-    }
+    } while ((delay == 0 ? Clock.lastKnownForwardProgressingMillis() : Clock.accurateForwardProgressingMillis()) - startTime <= delay && isRunning());
```

Comment:
Yea, there is technically a reason but I realize it's harder to understand so I can simplify